PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/springcloud/SpringCloudAppPropertyView.java

```diff
@@ -711,16 +739,20 @@ private void handleTextDataBinding(Object model, String propertyName, JTextField
         textField.setText(text);
     }
 
-    private void prepareViewModel(AppResourceInner app, DeploymentResourceInner deploy, String testUrl) {
+    private void prepareViewModel(AppResourceInner app, DeploymentResourceInner deploy) {
         try {
             if (app == null) {
-                updateModel(null);
                 return;
             }
             this.appResourceInner = app;
             this.deploymentResourceInner = deploy;
             SpringAppViewModel targetViewModel = new SpringAppViewModel();
-            targetViewModel.setTestUrl(testUrl);
+            try {
+                String clusterKey = testKeyCache.get(clusterId);
+                targetViewModel.setTestUrl(AzureSpringCloudMvpModel.getTestEndpointForApp(clusterKey, app.name()));
+            } catch (ExecutionException e) {
+                targetViewModel.setTestUrl(e.getMessage());
```

Comment:
What if e.getMessage() returns empty? Like NPE?

Reply:
testKeyCache.get will invoke AzureSpringCloudMvpModel.getPrimaryTestEndpoint(key), it will only return NPE if clusterId is null, actually  clusterId cannot be null.